Admin/AdminBuilderInterface.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Admin;
+
+use Sonata\AdminBundle\Datagrid\DatagridInterface;
+use Symfony\Component\Form\FormInterface;
+
+/**
+ * Builds admin. There is a dependency on the AdminInterface because
+ * this object holds useful object to deal with this task, but there is
+ * probably a better design.
```

Comment:
Agreed, and as long as the admin is huge it will be hard to go to that better design… but that's one good step forward :)